nilearn/reporting/_get_clusters_table.py

```diff
@@ -201,26 +206,6 @@ def get_clusters_table(stat_img, stat_threshold, cluster_threshold=None,
             )
             continue
 
-        # Extract connected components above cluster size threshold
-        label_map = ndimage.measurements.label(binarized, conn_mat)[0]
-        clust_ids = sorted(list(np.unique(label_map)[1:]))
-        for c_val in clust_ids:
-            if cluster_threshold is not None and np.sum(
-                    label_map == c_val) < cluster_threshold:
-                temp_stat_map[label_map == c_val] = 0
-                binarized[label_map == c_val] = 0
-
-        # If the cluster threshold is too high simply return an empty dataframe
-        # this checks for stats higher than threshold after small clusters
-        # were removed from temp_stat_map
-        if np.sum(temp_stat_map > stat_threshold) == 0:
-            warnings.warn(
-                'Attention: No clusters with more than {0} voxels'.format(
-                    cluster_threshold,
-                )
-            )
-            continue
-
```

Comment:
This initial threshold + label step is just to perform cluster-extent thresholding, which `threshold_img` can now do. The second step, in which the clusters are broken up and investigated individually, is still necessary.